### ISSUE TITLE
Row should be dict not list

### ISSUE THREAD

**Issue by saulshanabrook** [jazzband/tablib#158]
Would it make sense for rows to behave like dictionaries instead of lists? I would love to be able to do key based field retrieval, instead of index based.

**Comment by DanielDisisto** [comment]
PR #268 looks like a great solution, and for me at least would make tablib much more useful. Is this being considered for merge? Given the change to `Dataset.__getitem__` return type, I understand this is not a simple decision to make, but I think it would be beneficial.

If the change is considered too breaking, how about leaving `__getitem__` return a tuple as currently does, but adding a `Dataset.get_row(i)` which returns the `Row` instance (just to prevent having to access `Dataset._data` directly)? It's a bit clunky, but avoids all backward-compatibility issues while still providing a way to access row data by column name.

**Comment by ms8r** [comment]
Good point - I think we could get around the breaking issue in #268 by placing the logic of #268 behind the` Dataset.dict` API (less expensive than running `_package` each time) while continuing to return tuples for plain vanilla `__getitem__` access.  Will be a few weeks until I have some time for this again but will look into amending #268.
